ONSdigital/eq-survey-runner
# Add collection_metadata, make metadata immutable

### What is the context of this PR?
[Trello](https://trello.com/c/ET4X7qU4/2343-eq-survey-runner-1731-startedat-wiped-from-metadata-when-a-questionnaire-is-re-launched-issue)

Closes #1731 

Metadata is passed in and shouldn't be changed. It should be immutable.

We have metadata around the collection of the questionnaire which can't be immutable and doesn't fit in with the rest of `metadata` (started_at). Store it in `collection_metadata`.

`MappingProxyType` is used to make `metadata` a read-only view of `_metadata`. 

### How to review 
- Launch a questionnaire taking note of `RU Ref` and `Collection Exercise SID`
- Tap start survey and ensure `started_at` has been set.
- Launch with the same `RU Ref` and `Collection Exercise SID` and ensure `started_at` doesn't change.

### Checklist
